<?xml version="1.0"?>

<robot name="puzzle_drone">

   <link name="base_link">
        <visual>
            <origin xyz="0 0 -0.0205" rpy="1.57 0 1.57"/>
            <geometry>
                <mesh filename="package://link_act/meshes/base_210mm.stl"/>  
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>


   <link name="prop_1">
        <visual>
            <origin xyz="0 0 -0.004" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/propeller_ccw_puller_5in.stl"/>
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>

  <link name="prop_2">
        <visual>
            <origin xyz="0 0 -0.004" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/propeller_cw_puller_5in.stl"/>
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>

    <link name="prop_3">
        <visual>
            <origin xyz="0 0 -0.004" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/propeller_ccw_puller_5in.stl"/>
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>

    <link name="prop_4">
        <visual>
            <origin xyz="0 0 -0.004" rpy="0 0 0"/>
            <geometry>
                <mesh filename="package://link_act/meshes/propeller_cw_puller_5in.stl"/>
            </geometry>
            <material name="yellow">
                <color rgba="0.8 0.8 0.05 1.0"/>
            </material>
        </visual>
    </link>


    <joint name="prop1_joint" type="continuous">
        <parent link="base_link"/> 
        <child link="prop_1"/> 
        <origin xyz="0.06717 0.082 -0.0125" rpy="0 0 0"/>
        <axis xyz="0 0 1"/> 
    </joint>

    <joint name="prop2_joint" type="continuous">
        <parent link="base_link"/> 
        <child link="prop_2"/> 
        <origin xyz="-0.06717 0.082 -0.0125" rpy="0 0 0"/>
        <axis xyz="0 0 1"/> 
    </joint>

    <joint name="prop3_joint" type="continuous">
        <parent link="base_link"/> 
        <child link="prop_3"/> 
        <origin xyz="-0.06717 -0.082 -0.0125" rpy="0 0 0"/>
        <axis xyz="0 0 1"/> 
    </joint>

    <joint name="prop4_joint" type="continuous">
        <parent link="base_link"/> 
        <child link="prop_4"/> 
        <origin xyz="0.06717 -0.082 -0.0125" rpy="0 0 0"/>
        <axis xyz="0 0 1"/> 
    </joint>

</robot>